### DISCUSSION TITLE
WARNING - Connection schemes (type: google_cloud_platform) shall not contain '_' according to RFC3986.

### DISCUSSION THREAD

**Discussion by kuikeelc** [apache/airflow#34308] [General]
### Apache Airflow version

2.7.1

### What happened

When you use a connection with a type that holds an underscore (_), Airflow gives the following warning:

WARNING - Connection schemes (type: google_cloud_platform) shall not contain '_' according to RFC3986.

I don't think it is so bad that underscores are used. Can the warning be removed again?

### How to reproduce

No warning 

### Operating System

Mac OS X

### Versions of Apache Airflow Providers

independent of providers

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

none

### Anything else

No

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
You should explain WHERE you see it. RFC3986 refers to URI. If you see it in the URI, then yes it should not be there. Please describe how is that you get the warning - with details on what you do and how and more details/stacktraces etc. where you see it and then we can discuss it and see if it is justified or not.

But I believe this is a discussion, not issue, because most likely this is something that should be printed if URI is generated with such scheme

**Reply by potiuk** [reply]
For the future also - please note that "How to reproduce" section is there for a reason and  it's better to provide reproduction steps upfront, rather than being called to do so including converting your issue into discussion.

**Reply by kuikeelc** [reply]
The title holds the WHERE. It's in this part of the Airflow code and it complains about underscores in the connection name (e.g. google_cloud_default and it shows up in the logfiles during execution. It is in this part of the code:

https://airflow.apache.org/docs/apache-airflow/2.3.4/_modules/airflow/models/connection.html#Connection.get_uri

(look for RFC3986)

**Reply by potiuk** [reply]
What I missed my friend is WHERE it is generated from - you have forgotten to mention that it's the `get_uri` method. This (how crucial) piece of information was missing in the original report. So thank you for filling the gap.

**Reply by potiuk** [reply]
Converted it backt to issue, now when whe we know WHERE the issue gets generated.

**Comment by uranusjr** [comment]
RFC 3986 indeed does not allow underscore so the warning is legit. It works for urllib.parse but there’s no guarantee it works with other things, or will in the future. If anything I’d suggest start disallowing it entirely instead. Why is allowing the underscore important? Why can’t we switch to say `google-cloud-platform` instead?

**Reply by uranusjr** [reply]
Thinking about this a bit more, maybe we should just start sliently coercing underscores to dashes instead. Allowing (say) both `google-clould-platform` and `google_could_platform` as connection types seems like an unnecessary source of possible bugs that benefits mostly nobody.

**Comment by kuikeelc** [comment]
I don’t see what’s the use of this. Airflow has always been snake case and
there is no reason to change it.

My suggestion would be to leave it and focus on more important things.

Op di 24 okt 2023 om 06:31 schreef Tzu-ping Chung ***@***.***>

**Reply by potiuk** [reply]
See my comment in the issue created from that discussion @kuikeelc https://github.com/apache/airflow/issues/34419#issuecomment-1776787025 . If you have any comments to the suggestion and proposal I made there that aims to propose a solution and how to do it, feel free to comment there.

**Comment by potiuk** [comment]
Closing the discussion there as it is continued in #34419
